src/sentry/utils/concurrent.py

```diff
@@ -154,8 +155,8 @@ def __worker(self):
                 continue
             try:
                 result = function()
-            except Exception as error:
-                future.set_exception(error)
+            except Exception:
+                future.set_exception_info(*sys.exc_info()[1:])
```

Comment:
Nice, just out of curiosity, why do we cut out the first element from `.exc_info()`

Reply:
The first parameter is redundant (it can be derived from the second one): https://github.com/agronholm/pythonfutures/blob/5edbc65401fd578e195c2f0e2e8d3a6b5404f6bc/concurrent/futures/_base.py#L406-L414

I think the specific return type of `exc_info` is a legacy Python object model thing but I don't remember the specifics.